src/main/java/org/wso2/lsp4intellij/contributors/symbol/LineMarkerProvider.java

```diff
@@ -0,0 +1,225 @@
+      return;
+    }
+
+    final PsiFile containingFile = element.getContainingFile();
+    final VirtualFile virtualFile = containingFile.getVirtualFile();
+
+    if (!FileUtils.isFileSupported(virtualFile)) {
+      return;
+    }
+
+    // load data from server
+    final Set<LanguageServerWrapper> wrappers = ServiceManager.getService(IntellijLanguageClient.class).getAllServerWrappersFor(FileUtils.projectToUri(element.getProject()));
+    final Optional<LanguageServerWrapper> wrapperOpt = wrappers.stream().findFirst();
+    if (!wrapperOpt.isPresent()) {
+      return;
+    }
+    final LanguageServerWrapper wrapper = wrapperOpt.get();
+    final TextDocumentIdentifier textDocument = new TextDocumentIdentifier(FileUtils.uriFromVirtualFile(virtualFile));
+    final RequestManager requestManager = wrapper.getRequestManager();
+    if (requestManager == null) {
+      return;   // not connected
+    }
+    final CompletableFuture<List<Either<SymbolInformation, DocumentSymbol>>> listCompletableFuture = requestManager.documentSymbol(new DocumentSymbolParams(textDocument));
+
+    if (listCompletableFuture == null) {
+      return;
+    }
+
+    List<Either<SymbolInformation, DocumentSymbol>> eithers;
+    try {
+      eithers = listCompletableFuture.get(Timeout.getTimeout(Timeouts.SYMBOLS), TimeUnit.MILLISECONDS);
+      wrapper.notifySuccess(Timeouts.SYMBOLS);
+    } catch (InterruptedException | ExecutionException | TimeoutException e) {
+      wrapper.notifyFailure(Timeouts.SYMBOLS);
+      return;
+    }
+
+    if (eithers == null) {
+      return;
+    }
+
+    final Document document = FileDocumentManager.getInstance().getDocument(virtualFile);
+    if (document == null) {
+      return;
+    }
+    final Optional<Editor> foundEditor = Arrays.stream(EditorFactory.getInstance().getEditors(document, element.getProject())).findFirst();
+    if (!foundEditor.isPresent()) {
+      return;
+    }
+    Editor editor = foundEditor.get();
+
+    final Project project = element.getProject();
+    // iterate over documentSymbols
+    for (Either<SymbolInformation, DocumentSymbol> either : eithers) {
+      final LSPPsiSymbol lspPsiSymbol;
+      final Position startPos;
+      if (either.isLeft()) {
+        final SymbolInformation symbolInfo = either.getLeft();
+        if (symbolInfo.getKind() != SymbolKind.Method && symbolInfo.getKind() != SymbolKind.Constructor && symbolInfo.getKind() != SymbolKind.Class && symbolInfo.getKind() != SymbolKind.Interface) {
+          continue;
+        }
+        startPos = symbolInfo.getLocation().getRange().getStart();
+        final int start = DocumentUtils.LSPPosToOffset(editor, startPos);
+        final int end = DocumentUtils.LSPPosToOffset(editor, symbolInfo.getLocation().getRange().getEnd());
+        if (start < 0 || end < 0) {
+          continue;
+        }
+        lspPsiSymbol = new LSPPsiSymbol(symbolInfo.getKind(), symbolInfo.getName(), project, start, end, containingFile);
```

Comment:
I'm working on a similar FEature and was using your PR as a reference, Thank you :)

I solved this particular line differently (Kotlin code, but i think it's)
```
File containingFile = elements[0].containingFile;
PsiSymbol symbol = containingFile.findElementAt(start);

if (symbol == null || !elements.contains(symbol)) {
    continue
}
```

this way we don't need the check in line 79 (which was not working for me..) and intellij reacts quickly on user input